server/spec/refresh_pools_spec.rb

```diff
@@ -217,10 +214,14 @@
     product2 = create_product(name, name, :owner => owner2['key'])
 
     # migrate the subscription to another owner.
+    sub = get_hostedtest_subscription(pool.subscriptionId)
     sub["owner"] = owner2
-    @cp.update_subscription(sub)
-    @cp.list_subscriptions(owner1["key"]).length.should == 0
-    @cp.list_subscriptions(owner2["key"]).length.should == 1
+    update_hostedtest_subscription(sub)
+    #TODO - removing sleeps gives concurrent modification errrors - investigate
```

Comment:
@Ceiu yes, in candlepin . 
I'd recommend tackling this in a separate PR. I might be wrong , but I think this is already broken in master.


org=] INFO  org.candlepin.resource.OwnerResource - Cleaning up owner: Owner [id: ff8080815111ba3d015111bd259a023d, key: initial-owner-wi0Hxy90]
2015-11-16 14:19:48,852 [thread=QuartzScheduler_Worker-11] [job=refresh_pools_f5d4845b-bd06-41a8-840e-445f372b0325, org=] ERROR org.candlepin.pinsetter.tasks.RefreshPoolsJob - RefreshPoolsJob encountered a problem.
org.candlepin.common.exceptions.ConcurrentModificationException: Request failed due to concurrent modification, please re-try.
        at org.candlepin.model.AbstractHibernateCurator.flush(AbstractHibernateCurator.java:386) ~[AbstractHibernateCurator.class:na]
        at org.candlepin.model.PoolCurator.delete(PoolCurator.java:559) ~[PoolCurator.class:na]
        at com.google.inject.persist.jpa.JpaLocalTxnInterceptor.invoke(JpaLocalTxnInterceptor.java:58) ~[guice-persist-3.0.jar:na]
        at org.candlepin.controller.CandlepinPoolManager.deletePool(CandlepinPoolManager.java:1590) ~[CandlepinPoolManager.class:na]
        at com.google.inject.persist.jpa.JpaLocalTxnInterceptor.invoke(JpaLocalTxnInterceptor.java:58) ~[guice-persist-3.0.jar:na]
        at org.candlepin.controller.CandlepinPoolManager.removeAndDeletePoolsOnOtherOwners(CandlepinPoolManager.java:563) ~[CandlepinPoolManager.class:na]
        at org.candlepin.controller.CandlepinPoolManager.refreshPoolsForSubscription(CandlepinPoolManager.java:502) ~[CandlepinPoolManager.class:na]
        at com.google.inject.persist.jpa.JpaLocalTxnInterceptor.invoke(JpaLocalTxnInterceptor.java:66) ~[guice-persist-3.0.jar:na]
        at org.candlepin.controller.CandlepinPoolManager.refreshPoolsWithRegeneration(CandlepinPoolManager.java:192) ~[CandlepinPoolManager.class:na]